src/hl_memory.c

```diff
@@ -6,101 +6,171 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
+#include <pthread.h>
+#include <fcntl.h>
 #include "hl_memory.h"
 #include "perftest_parameters.h"
 #include "synapse_api.h"
```

Comment:
is here only for `synDeviceId`, if the member will be removed, remove this include as well